llvm/lib/Support/Triple.cpp

```diff
@@ -261,6 +261,7 @@ Triple::ArchType Triple::getArchTypeForLLVMName(StringRef Name) {
     .Case("aarch64_be", aarch64_be)
     .Case("arc", arc)
     .Case("arm64", aarch64) // "arm64" is an alias for "aarch64"
+    .Case("arm64_32", aarch64) // "arm64" is an alias for "aarch64"
```

Comment:
Why does this not have its own `arch`?  That seems extremely bug-prone for all clients of the triple.

Reply:
That sounds like a really good idea, but it didn't occur to me at the time.

Originally I favoured aarch64-apple-watchos-code32, but that caused problems with LTO and the binutilsy tools which only wanted to look at the architecture part of the triple.

A new arch here shouldn't suffer from that, though we'll probably need to update lldb & ld64 (and Swift of course).

I'll start a prototype.

Reply:
Thanks!  I assume that arch will also be considered 32-bit?

Reply:
Yep, I implemented it today and a large part of the benefit IMO has been that isArch32Bit is now available straight from the Triple for when you want to check for ILP32.

Reply:
Awesome!  That'll knock out a number of special cases we've had to track down and add in various places.